platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PreconsensusEventReplayWorkflow.java

```diff
@@ -0,0 +1,245 @@
+     * @param consensusRoundHandler              the object responsible for applying transactions to consensus rounds
+     * @param stateHashSignQueue                 the queue thread for hashing and signing states
+     * @param stateManagementComponent           manages various copies of the state
+     * @param currentPlatformStatus              a pointer to the current platform status
+     * @param initialMinimumGenerationNonAncient the minimum generation of events to replay
+     */
+    public static void replayPreconsensusEvents(
+            @NonNull final PlatformContext platformContext,
+            @NonNull final ThreadManager threadManager,
+            @NonNull final Time time,
+            @NonNull final PreconsensusEventFileManager preConsensusEventFileManager,
+            @NonNull final PreconsensusEventWriter preConsensusEventWriter,
+            @NonNull final EventTaskDispatcher eventTaskDispatcher,
+            @NonNull final EventIntake eventIntake,
+            @NonNull final QueueThread<EventIntakeTask> intakeQueue,
+            @NonNull final ConsensusRoundHandler consensusRoundHandler,
+            @NonNull final QueueThread<ReservedSignedState> stateHashSignQueue,
+            @NonNull final StateManagementComponent stateManagementComponent,
+            @NonNull final AtomicReference<PlatformStatus> currentPlatformStatus,
+            final long initialMinimumGenerationNonAncient) {
+
+        Objects.requireNonNull(platformContext);
+        Objects.requireNonNull(time);
+        Objects.requireNonNull(preConsensusEventFileManager);
+        Objects.requireNonNull(preConsensusEventWriter);
+        Objects.requireNonNull(eventTaskDispatcher);
+        Objects.requireNonNull(eventIntake);
+        Objects.requireNonNull(intakeQueue);
+        Objects.requireNonNull(consensusRoundHandler);
+        Objects.requireNonNull(stateHashSignQueue);
+        Objects.requireNonNull(stateManagementComponent);
+        Objects.requireNonNull(currentPlatformStatus);
+
+        setupReplayStatus(currentPlatformStatus);
+
+        logger.info(
+                STARTUP.getMarker(),
+                "replaying preconsensus event stream starting at generation {}",
+                initialMinimumGenerationNonAncient);
+
+        try {
+            final Instant start = time.now();
+
+            final IOIterator<EventImpl> iterator =
+                    preConsensusEventFileManager.getEventIterator(initialMinimumGenerationNonAncient, true);
+
+            final PreconsensusEventReplayPipeline eventReplayPipeline = new PreconsensusEventReplayPipeline(
+                    platformContext, threadManager, iterator, eventTaskDispatcher::dispatchTask);
+            eventReplayPipeline.replayEvents();
+
+            waitForReplayToComplete(intakeQueue, consensusRoundHandler, stateHashSignQueue);
+
+            final Instant finish = time.now();
+            final Duration elapsed = Duration.between(start, finish);
+
```

Comment:
shouldnt we turn on state signing at this point?

Reply:
This is currently one of the very few things that is currently controlled by the platform status. So no need to do anything directly to enable.

```
        if (platformStatus != PlatformStatus.ACTIVE && platformStatus != PlatformStatus.MAINTENANCE) {
            // Only send transactions if the platform is in ACTIVE or MAINTENANCE state.
            return;
        }
```